modules/mini-games/Race.lua

```diff
@@ -370,32 +375,36 @@ end
 --- Make the car indestructible and cause all biters to flee
 local function start_invincibility(car,name)
     car.destructible = false
-    local biters = surface[1].find_enemy_units(variables["Dead_car"][name].position, 50, "player")
+    local biters = surface[1].find_enemy_units(dead_cars[name].position, 50, "player")
     for i, biter in ipairs(biters) do
         biter.set_command{ type = defines.command.flee, distraction = defines.distraction.by_anything, from = car }
     end
 end
 
 --- Make the car and player destructible again
 local stop_invincibility = Token.register(function(name)
-    variables["Dead_car"][name].car.destructible = true
-    local player = variables["Dead_car"][name].player
+    dead_cars[name].car.destructible = true
+    local player = dead_cars[name].player
     player.character.destructible = true
 end)
 
 --- Kill all biters in a close range to the player
 local kill_biters = Token.register(function(name)
-    local biters = surface[1].find_enemy_units(variables["Dead_car"][name].position, 3, "player")
+    local biters = surface[1].find_enemy_units(dead_cars[name].position, 3, "player")
     for i, biter in ipairs(biters) do
         biter.destroy()
     end
 end)
 
 --- Respawn the car for a player
-local respawn_car = Token.register(function(name)
-    local player = variables["Dead_car"][name].player
-    local position = surface[1].find_non_colliding_position('car', variables["Dead_car"][name].position, 5, 0.5)
-    position = position or variables["Dead_car"][name].position
+local respawn_car
+respawn_car = Token.register(function(name)
+    local player = dead_cars[name].player
+    local position = surface[1].find_non_colliding_position('car', dead_cars[name].position, 5, 0.5)
+    if not position then
+        local offset = math.random(-10, 10)
+        return task.set_timeout_in_ticks(30+offset, respawn_car, name)
+    end
```

Comment:
the random here is 100% useless as if a position cant be find it will just be ran again anyway

Reply:
Removed.